src/components/textarea/daikin-textarea.ts

```diff
@@ -137,69 +132,58 @@ export class DaikinTextarea extends LitElement {
   autocomplete?: HTMLInputElement["autocomplete"];
 
   /**
-   * Error state. Ignored if the `disabled` is `true`.
+   * Whether to allow resizing of the text area
    */
   @property({ type: Boolean, reflect: true })
-  error = false;
+  resizable = false;
 
-  /**
-   * Whether to display a counter at the top of the text area
-   */
-  @property({ type: Boolean, reflect: true })
-  counter = false;
-
-  @state()
-  protected _textareaCounter: number = 0;
-
-  private _updateValueAndCounter(value: string): void {
+  private _updateValue(value: string): void {
     this._internals.setFormValue(value);
-    this._textareaCounter = [...value].length;
   }
 
   private _handleInput(e: InputEvent): void {
     this.value = (e.target as HTMLInputElement).value;
-    this._updateValueAndCounter(this.value);
+    this._updateValue(this.value);
+
+    this.dispatchEvent(
+      new CustomEvent("change-count", {
+        detail: {
+          count: this.value.length,
+        },
+        bubbles: true,
+        composed: true,
+        cancelable: false,
+      })
+    );
   }
 
   override render() {
-    const textareaClassName = cvaTextarea({
-      variant: !this.disabled && this.error ? "error" : "normal",
-    });
-
-    const textareaCounterClassName = cvaCounter({
-      variant: this.disabled ? "disabled" : "normal",
-    });
-
     return html`<textarea
-        class=${textareaClassName}
-        .value=${this.value}
-        placeholder=${this.placeholder}
-        maxlength=${this.maxlength}
-        autocomplete=${
-          // eslint-disable-next-line @typescript-eslint/no-explicit-any -- workaround lit-analyzer checking
-          ifDefined(this.autocomplete as any)
-        }
-        ?disabled=${this.disabled}
-        ?readonly=${this.readonly}
-        @change=${(e: Event) => this.dispatchEvent(new Event("change", e))}
-        @input=${this._handleInput}
-        @keydown=${this._handleInput}
-      ></textarea>
-      ${this.counter
-        ? html`
-            <span class=${textareaCounterClassName}
-              >${this._textareaCounter}/${this.maxlength}</span
-            >
-          `
-        : null}`;
+      class=${cvaTextarea({
+        status: !this.disabled && this.error ? "error" : "default",
+        resize: this.resizable,
+      })}
+      .value=${this.value}
+      placeholder=${this.placeholder}
+      autocomplete=${
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any -- workaround lit-analyzer checking
+        ifDefined(this.autocomplete as any)
+      }
+      ?disabled=${this.disabled}
+      ?readonly=${this.readonly}
+      aria-required=${this.required}
```

Comment:
I think we should use `required` attribute.

```suggestion
      ?required=${this.required}
```